### DISCUSSION TITLE
Postgresql-based search doesn't include unpublished pages when reconstructing index

### DISCUSSION THREAD

**Discussion by hugaleno** [requarks/wiki#6212] [Error / Bug Report] [answered]
We use wiki js to document internal procedures in the company I work. For security purpose, we require all users to be autenticated to see the documents. To achieve that, we unpublish all pages.

When reconstructing the postgresql advanced search index, the content of unpublished pages are not included. Because of that the pages are not found in the search.

If this is a intend behavior, maybe turn this in a configuration option.

Update: Just realized that we are using the wrong mecanism to require user authentication, now doing the right way using "guests" group permission. Maybe the rest still apply

**Comment by fnafox** [comment, marked as answer]
This doesn't seem like the intended use of the publish feature to me. It's meant for working on a page without releasing it to everyone until it's ready.
What you want can be accomplished by setting up permissions and disallowing unauthenticated users (guest) from accessing content.
If you want to require authentication for all users, revoke read permissions for the guest group.

**Reply by hugaleno** [reply]
Yes, you right. I used that to require authentication to access the pages and published all pages again.
Was using the wrong feature to achieve that.
